src/AI.cpp

```diff
@@ -0,0 +1,184 @@
+
+    maxReconnect = additionalOptions.find("maxReconnect") != additionalOptions.end() ? std::stoi(
+            additionalOptions.at("maxReconnect")) : 5;
+
+    // set up logging
+    Logging::initLogging(verbosity);
+
+    connect();
+}
+
+void AI::connect() {
+    // connect to server as AI
+    if (!libClientHandler.network.connect(address, port)) {
+        spdlog::critical("could not connect to server");
+        exit(2);
+    }
+    spdlog::info("connected to server");
+    if (!libClientHandler.network.sendHello(name, spy::network::RoleEnum::AI)) {
+        spdlog::critical("could not send Hello message");
+        exit(2);
+    }
+    spdlog::info("send Hello message");
+
+    // TODO make sure that matchConfig is available
+}
+
+void AI::onHelloReply() {
+    spdlog::info("received HelloReply message");
+    // do nothing
+}
+
+void AI::onGameStarted() {
+    spdlog::info("received GameStarted message");
+    // do nothing
+}
+
+void AI::onRequestItemChoice() {
+    spdlog::info("received RequestItemChoice message");
+
+    auto choice = ItemChoice::generate(difficulty, libClientHandler.getOfferedCharacters(),
+                                       libClientHandler.getOfferedGadgets(), matchConfig);
+    if (!libClientHandler.network.sendItemChoice(choice)) {
+        spdlog::critical("could not send ItemChoice message");
+        exit(2);
+    }
+    spdlog::info("sent ItemChoice message");
+}
+
+void AI::onRequestEquipmentChoice() {
+    spdlog::info("received RequestEquipmentChoice message");
+
+    auto equipment = EquipmentChoice::generate(difficulty, libClientHandler.getChosenCharacters(),
+                                               libClientHandler.getChosenGadgets(), matchConfig);
+    if (!libClientHandler.network.sendEquipmentChoice(equipment)) {
+        spdlog::critical("could not send EquipmentChoice message");
+        exit(2);
+    }
+    spdlog::info("sent EquipmentChoice message");
+}
+
+void AI::onGameStatus() {
+    spdlog::info("received GameStatus message");
+
+    // TODO save additional info (can this be done in LibClient to help Client too) ?
+}
+
+void AI::onRequestGameOperation() {
+    spdlog::info("received RequestGameOperation message");
+
+    auto operation = GameOperation::generate(difficulty, libClientHandler.getActiveCharacter(),
+                                             libClientHandler.getState(), matchConfig);
+    if (!libClientHandler.network.sendGameOperation(operation, matchConfig)) {
+        spdlog::critical("could not send GameOperation message");
+        exit(2);
+    }
+    spdlog::info("sent GameOperation message");
+}
+
+void AI::onStatistics() {
+    spdlog::info("received Statistics message");
+
+    // TODO use as additional info ?
+    std::string endString = "Game ";
+    endString += libClientHandler.getWinner().value() == libClientHandler.getId().value() ? "won!" : "lost!";
+    spdlog::info(endString);
+    libClientHandler.network.disconnect();
```

Comment:
das sollte im destruktor in der libclient geschehen

Reply:
Ja, stimmt. Das ist redundant.

Reply:
Finde ich das nur nicht oder ist das da noch nicht drin? ist nicht so super relevant, man könnte ein issue dafür aufmachen

Reply:
default destructor reicht aus